app/src/main/java/org/mozilla/fenix/perf/ProfilerUtils.kt

```diff
@@ -0,0 +1,163 @@
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.fenix.perf
+
+import android.content.ClipData
+import android.content.ClipboardManager
+import android.content.Context
+import android.content.Intent
+import android.os.Build
+import android.provider.DocumentsContract
+import android.util.Base64
+import android.widget.Toast
+import androidx.activity.result.ActivityResultLauncher
+import mozilla.components.concept.fetch.MutableHeaders
+import mozilla.components.concept.fetch.Request
+import mozilla.components.concept.fetch.Response
+import org.json.JSONObject
+import org.mozilla.fenix.R
+import org.mozilla.fenix.ext.components
+import java.io.BufferedReader
+import java.io.ByteArrayInputStream
+import java.io.File
+import java.io.FileOutputStream
+import java.io.InputStreamReader
+import java.util.zip.GZIPInputStream
+
+private val profiler_api = "https://api.profiler.firefox.com/compressed-store"
+private val profiler_header = "application/vnd.firefox-profiler+json;version=1.0"
+private val token = "profileToken"
+private val profileDataURL = "https://profiler.firefox.com/public/"
+
+private val firefox_features = arrayOf("screenshots", "js", "leaf", "stackwalk", "cpu", "java", "processcpu", "ipcmessages", "java")
+private val firefox_threads = arrayOf("GeckoMain", "Compositor", "Renderer", "SwComposite", "DOM Worker",)
+
+private val graphics_features = arrayOf("java", "ipcmessages")
+private val graphics_threads = arrayOf("GeckoMain", "Compositor", "Renderer", "SwComposite", "RenderBackend", "SceneBuilder", "WrWorker", "CanvasWorkers",)
+
+private val media_features = arrayOf("js", "leaf", "stackwalk", "cpu", "audiocallbacktracing", "ipcmessages", "processcpu", "java")
+private val media_threads = arrayOf("cubeb", "audio", "camera", "capture", "Compositor", "GeckoMain", "gmp", "graph", "grph", "InotifyEventThread", "IPDL Background", "media", "ModuleProcessThread", "PacerThread", "RemVidChild", "RenderBackend", "Renderer", "Socket Thread", "SwComposite", "webrtc")
+
+private val networking_features = arrayOf("screenshots", "js", "leaf", "stackwalk", "cpu", "java", "processcpu", "ipcmessages")
+private val networking_threads = arrayOf("Compositor", "DNS Resolver", "DOM Worker", "GeckoMain", "Renderer", "Socket Thread", "StreamTrans", "SwComposite", "TRR Background")
+
+var isProfilerActive: Boolean = false
+
+enum class ProfilerSettings(settingType: String, threads: Array<String>, features: Array<String>) {
+    Firefox("Firefox", firefox_threads, firefox_features),
+    Graphics("Graphics", graphics_threads, graphics_features),
+    Media("Media", media_threads, media_features),
+    Networking("Firefox", networking_threads, networking_features);
+
+    var settingType: String? = settingType
+    var features: Array<String>? = features
+    var threads: Array<String>? = threads
+}
+
+fun saveProfileAsURL(profileResult: ByteArray, context: Context): String {
```

Comment:
nitpick: these shouldn't be defined at the top-level because it pollutes the global namespace. Can you put these in an `object`?